src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -660,11 +660,18 @@ protected override StatementSyntax CreateAssignmentExpressionStatement(SyntaxTok
                 var type = variable.GetVariableType();
                 var typeNode = type.GenerateTypeSyntax();
 
+                var originalIdentifierToken = variable.GetOriginalIdentifierToken(cancellationToken);
+                var usingKeyword = originalIdentifierToken.GetAncestor<LocalDeclarationStatementSyntax>() is LocalDeclarationStatementSyntax { UsingKeyword.FullSpan.IsEmpty: false }
```

Comment:
the unbounded walk up to any higher LocalDecl is fairly unsafe.  consider something unlikely, but possible, like:

```using var x = new Blah(() => { ... code of interest is all in here ... })```

Generally speaking, you want to walk up very judiciously.  IN this case, likely just seeing if you are parented directly by a vardeclarator, then vardeclaration, then finally teh localdeclstatement.

(yes, it's unlikely... but weird things do happen and unbounded looking upwards does bite us :)).